File: src/main/java/com/app/umkaSchool/service/GroupService.java
package com.app.umkaSchool.service;

import com.app.umkaSchool.dto.group.CreateGroupRequest;
import com.app.umkaSchool.dto.group.GroupResponse;
import com.app.umkaSchool.dto.group.UpdateGroupRequest;
import com.app.umkaSchool.model.StudentGroup;

import java.util.List;
import java.util.UUID;

public interface GroupService {
    GroupResponse createGroup(CreateGroupRequest request);

    GroupResponse updateGroup(UUID groupId, UpdateGroupRequest request);

    GroupResponse getGroupById(UUID groupId);

    GroupResponse getGroupByCode(String code);

    List<GroupResponse> getAllGroups();

    List<GroupResponse> getGroupsByTeacher(UUID teacherId);

    void deleteGroup(UUID groupId);

    void addStudentsToGroup(UUID groupId, List<UUID> studentIds);

    void removeStudentFromGroup(UUID studentId);

    StudentGroup getGroupEntity(UUID groupId);
}

File: frontend/src/services/group.service.ts
import axiosInstance from './axios.config';
import { Group, CreateGroupPayload, UpdateGroupPayload } from '../types/group';

export const groupService = {
    getAllGroups: async (): Promise<Group[]> => {
        const response = await axiosInstance.get<Group[]>('/api/groups');
        return response.data;
    },

    getGroupById: async (groupId: string): Promise<Group> => {
        const response = await axiosInstance.get<Group>(`/api/groups/${groupId}`);
        return response.data;
    },

    getGroupByCode: async (code: string): Promise<Group> => {
        const response = await axiosInstance.get<Group>(`/api/groups/code/${code}`);
        return response.data;
    },

    getGroupsByTeacher: async (teacherId: string): Promise<Group[]> => {
        const response = await axiosInstance.get<Group[]>(`/api/groups/teacher/${teacherId}`);
        return response.data;
    },

    createGroup: async (payload: CreateGroupPayload): Promise<Group> => {
        const response = await axiosInstance.post<Group>('/api/groups', payload);
        return response.data;
    },

    updateGroup: async (groupId: string, payload: UpdateGroupPayload): Promise<Group> => {
        const response = await axiosInstance.put<Group>(`/api/groups/${groupId}`, payload);
        return response.data;
    },

    deleteGroup: async (groupId: string): Promise<void> => {
        await axiosInstance.delete(`/api/groups/${groupId}`);
    },

    addStudentsToGroup: async (groupId: string, studentIds: string[]): Promise<void> => {
        await axiosInstance.post(`/api/groups/${groupId}/students`, studentIds);
    },

    removeStudentFromGroup: async (studentId: string): Promise<void> => {
        await axiosInstance.delete(`/api/groups/students/${studentId}`);
    }
};


